src/rtcan.c

```diff
@@ -209,6 +189,54 @@ rtcan_status_t rtcan_init(rtcan_handle_t* rtcan_h,
  */
 rtcan_status_t rtcan_start(rtcan_handle_t* rtcan_h)
 {
+    /* Since the subscriber ID list is known, we can configure the CAN ID Filter now */
+
+    uint32_t can_id_list[RTCAN_HASHMAP_SIZE] = {0};
+    int number_ids = 0;
+
+    /* Save and count all CAN IDs stored in hashmap */
+    for (uint32_t i = 0; i < RTCAN_HASHMAP_SIZE; i++)
+    {
+        if(rtcan_h->subscriber_map[i] != NULL)
+        {
+            /* Subscriber found! */
+            can_id_list[number_ids] = rtcan_h->subscriber_map[i]->can_id;
+            number_ids++;
+
+        }
+    }
+
+    int number_banks = (number_ids-1)/4 + 1;
+
+    /* Error if there's too many ids to apply filter */
+    ADD_ERROR_IF(number_banks > 28, RTCAN_ERROR_INIT, rtcan_h);
```

Comment:
Where does the 28 come from? I feel like it would be different depending on the STM32 in question and this code will be deployed to F7 and F4 boards. Either way it should be a constant, and probably a user-specified one? Also worth checking if it exists as part of the HAL.

Reply:
28 is the number of filter banks and for F7 and F4 is the same. It's not defined anywhere in HAL unfortunately.